src/guide/advanced/intro.md

```diff
@@ -96,17 +96,16 @@ edition = "2021"
 crate-type = ['cdylib']
 
 [dependencies]
-iroha_data_model = { git = "https://github.com/hyperledger/iroha/", branch = "iroha2", default-features = false }
 iroha_wasm = { git = "https://github.com/hyperledger/iroha/", branch = "iroha2" }
 
 parity-scale-codec = {version = "2.3.1", default-features = false }
 ```
 
-Note that the crate type is `cdylib`. WASM is a portable format, most Rust code is linked in a non-portable architecture- and OS-specific static manner. As Rust does not (at present) have a stable ABI, we have to rely on the C-linkage to generate WASM. You don't have to worry about C foreign function interfaces too much in the smart contract itself, we took care of most of that for you.
+Note that the crate type is `cdylib`. Most Rust code is linked in a non-portable architecture and OS-specific static manner, but WASM is a portable format. Since C ABI is the lingua franca of the programming world and there is no other stable Rust ABI, Iroha relies on the C-linkage to generate WASM bindings. Note that you don't have to worry about C FFI in the smart contract itself, most of that is taken care for you.
 
-`iroha_data_model` is the crate that contains the basic ISI and types. `iroha_wasm` is the crate that contains all of the bindings, macros and trait implementations that you'd need to write the program. `parity-scale-codec` is (as of today) the chosen serialisation format. We intend to replace it with a different format in the near future, as this particular crate is rather large[^2].
+`iroha_wasm` is the crate that contains all of the bindings, macros and trait implementations that you'd need to write the program, most notably the `iroha_wasm` attribute macro. This crate also exposes `data_model` which contains all of the basic ISI and types. `parity-scale-codec` is (as of today) the chosen serialisation format with a strong possibility of getting replaced by a different(custom) serialization format in the near future, as it seems to dominate the binary size[^2].
 
-Now that we have the preliminaries nailed down, we get to writing some code. In the `lib.rs` you should write the following:
+Now that the preliminaries are nailed down, we get to write a basic smart contract. In the `src/lib.rs` you should write the following:
```

Comment:
```suggestion
Now that we have the preliminaries nailed down, we get to write some code for our smart contract. In the `src/lib.rs` you should write the following:
```
The first is a stylistic choice. 

The second is emphasising that so far what we've been doing is a smart contract, just not source code.